samples/GalleryApp/ObjectDetectorPage.xaml.cs

```diff
@@ -0,0 +1,494 @@
+        /// <param name="sender"></param>
+        /// <param name="frame"></param>
+        private void FrameSource_FrameAvailable(object sender, VideoFrame frame)
+        {
+            // Locking behavior, so only one skill execution happens at a time
+            if (m_lock.Wait(0))
+            {
+#pragma warning disable CS4014
+                // Purposely don't await this: want handler to exit ASAP
+                // so that realtime capture doesn't wait for completion.
+                // Instead, we unlock only when processing finishes ensuring that
+                // only one execution is active at a time, dropping frames or
+                // aborting skill runs as necessary
+                Task.Run(async () =>
+                {
+                    try
+                    {
+                        await DetectObjectsAsync(frame);
+                        await DisplayFrameAndResultAsync(frame);
+                    }
+                    catch (Exception ex)
+                    {
+                        NotifyUser(ex.Message);
+                    }
+                    finally
+                    {
+                        m_lock.Release();
+                    }
+                });
+#pragma warning restore CS4014
+            }
+        }
+
+        /// <summary>
+        /// Click handler for video file button. Spawns file picker UI
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private async void UIFilePickerButton_Click(object sender, RoutedEventArgs e)
+        {
+            // Disable the top menu while handling the click
+            await UpdateMediaSourceButtonsAsync(false);
+
+            var picker = new Windows.Storage.Pickers.FileOpenPicker();
+            picker.ViewMode = Windows.Storage.Pickers.PickerViewMode.Thumbnail;
+            picker.SuggestedStartLocation = Windows.Storage.Pickers.PickerLocationId.PicturesLibrary;
+            // Add common video file extensions
+            picker.FileTypeFilter.Add(".mp4");
+            picker.FileTypeFilter.Add(".avi");
+            // Add common image file extensions
+            picker.FileTypeFilter.Add(".jpg");
+            picker.FileTypeFilter.Add(".png");
+            picker.FileTypeFilter.Add(".bmp");
+
+            Windows.Storage.StorageFile file = await picker.PickSingleFileAsync();
+            if (file != null)
+            {
+                await ConfigureFrameSourceAsync(file);
+                NotifyUser("Loading file: " + file.Path);
+            }
+
+            // Re-enable the top menu once done handling the click
+            await UpdateMediaSourceButtonsAsync(true);
+        }
+
+        /// <summary>
+        /// Click handler for camera button. Spawns device picker UI
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private async void UICameraButton_Click(object sender, RoutedEventArgs e)
```

Comment:
>UICameraButton_Click [](start = 27, length = 20)

same here, please refactor the code here to be used by all skills. Similarly for the rest of the common UI.

Reply:
Will do refactor work in a new task (same as above comment).

---
In reply to: [297795414](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/39#discussion_r297795414) [](ancestors = 297795414)